Shadow Style

ssorallen opened the pull request:
Move styles inline to work in shadow DOM
- Remove tag name selector where scoping is no longer needed
- Replace tag name selector with `:host` when selecting based on host
  class name

Also...
- Add content to `<button>` elements to prevent them from collapsing in
  Firefox

Commit:
Move demo inside demo dir per repo design doc

Commit:
Outdent contents of `<html>` to match style guide

Commit:
Add content to `<button>` to not collapse in Firefox

`<button>` elements with no content collapse to weird gray blobs in
Firefox.

Commit:
Move styles inline to work in shadow DOM

* Remove tag name selector where scoping is no longer needed
* Replace tag name selector with `:host` when selecting based on host
  class name

morethanreal commented on core-toolbar.html:
```diff
@@ -59,19 +59,114 @@
+     * expected.
+     */
+    .toolbar-tools > *:not([disabled]),
+    ::content > *:not([disabled]) {
```
Is `toolbar-tools` a class set in the local or light children? It looks like `.toolbar-tools > *:not([disabled])` won't match anything.

ssorallen replied:
Aha thanks for that. This is the point you mentioned yesterday about wrapping `<content>` in another container.

morethanreal commented on index.html:
```diff
@@ -8,15 +8,15 @@
 
-    <script src="../webcomponentsjs/webcomponents.js"></script>
-    <link rel="import" href="../core-component-page/core-component-page.html">
+  <script src="../webcomponentsjs/webcomponents.js"></script>
```
`webcomponents-lite.js` since we don't use the shadow DOM polyfill?

ssorallen replied:
:+1:

Commit:
Use wrapper selector for content to work in shady

Per the Primer on [scoped styling](1):

> styling elements distributed to content (via ::content) requires
> using a wrapper element for compatibility with shady DOM

[1] https://github.com/Polymer/polymer/blob/0.8-preview/PRIMER.md#scoped-styling

Commit:
Prefer webcomponents lite since no shadow DOM polyfill

ssorallen commented:
@morethanreal, can you take another look?

morethanreal commented:
lgtm

ssorallen commented:
Thanks. :bee:
